SerialPrograms/Source/PokemonSV/Programs/ItemPrinter/PokemonSV_ItemPrinterRNG.cpp

```diff
@@ -697,6 +705,377 @@ void ItemPrinterRNG::run_print_at_date(
+        target_result_minus_1 = {"nest-ball", "poke-ball", "heal-ball", "premier-ball", "premier-ball", "", "", "", "", ""};
+        target_result         = {"moon-ball", "moon-ball", "moon-ball", "moon-ball", "moon-ball", "", "", "", "", ""};
+        target_result_plus_1  = {"fast-ball", "heal-ball", "poke-ball", "dusk-ball", "luxury-ball", "", "", "", "", ""};
+        target_result_plus_2  = {"lure-ball", "heavy-ball", "luxury-ball", "heal-ball", "poke-ball", "", "", "", "", ""};
+        break;  
+    case 1244233031:
+        current_print = "SPORT_BALL: ";
+        target_result_minus_2 = {"sport-ball", "heal-ball", "luxury-ball", "great-ball", "heal-ball", "", "", "", "", ""};
+        target_result_minus_1 = {"sport-ball", "dream-ball", "level-ball", "poke-ball", "luxury-ball", "", "", "", "", ""};
+        target_result         = {"sport-ball", "sport-ball", "sport-ball", "sport-ball", "sport-ball", "", "", "", "", ""};
+        target_result_plus_1  = {"moon-ball", "heal-ball", "poke-ball", "poke-ball", "heal-ball", "", "", "", "", ""};
+        target_result_plus_2  = {"moon-ball", "poke-ball", "ultra-ball", "poke-ball", "great-ball", "", "", "", "", ""};
+        break;    
+    case 2657784681:
+        current_print = "DREAM_BALL: ";
+        target_result_minus_2 = {"master-ball", "poke-ball", "luxury-ball", "level-ball", "dream-ball", "", "", "", "", ""};
+        target_result_minus_1 = {"timer-ball", "beast-ball", "poke-ball", "premier-ball", "heal-ball", "", "", "", "", ""};
+        target_result         = {"dream-ball", "dream-ball", "dream-ball", "dream-ball", "dream-ball", "", "", "", "", ""};
+        target_result_plus_1  = {"heavy-ball", "poke-ball", "friend-ball", "fast-ball", "heavy-ball", "", "", "", "", ""};
+        target_result_plus_2  = {"luxury-ball", "poke-ball", "luxury-ball", "quick-ball", "friend-ball", "", "", "", "", ""};
+        break;  
+    case 2117201835:
+        current_print = "BEAST_BALL: ";
+        target_result_minus_2 = {"luxury-ball", "poke-ball", "lure-ball", "master-ball", "dive-ball", "", "", "", "", ""};
+        target_result_minus_1 = {"heavy-ball", "premier-ball", "premier-ball", "poke-ball", "level-ball", "", "", "", "", ""};
+        target_result         = {"beast-ball", "beast-ball", "beast-ball", "beast-ball", "beast-ball", "", "", "", "", ""};
+        target_result_plus_1  = {"heal-ball", "great-ball", "luxury-ball", "heavy-ball", "great-ball", "", "", "", "", ""};
+        target_result_plus_2  = {"poke-ball", "premier-ball", "great-ball", "poke-ball", "dive-ball", "", "", "", "", ""};
+        break;       
+    default:
+        unknown_seed = true;
+        console.log("Unknown seed. Can't calculate distance from target.");
+        break;
+    }
+
+    if (unknown_seed){
+        return DistanceFromTarget::UNKNOWN;
+    }
+
+    if (print_results == target_result){

```

Comment:
I wonder if we need any leeway for misreads. So rather than requiring that everything matches, we perhaps allow maybe 1 mismatch, but only if there's at least N ingredients in the first place.

It may not be an issue for English, but I can see Chinese being an issue.